libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -175,24 +175,24 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 {
 	yulAssert(m_scope, "");
 
-	int const numVariables = _varDecl.variables.size();
-	int heightAtStart = m_assembly.stackHeight();
+	size_t const numVariables = _varDecl.variables.size();
+	auto heightAtStart = static_cast<size_t>(m_assembly.stackHeight());
 	if (_varDecl.value)
 	{
 		std::visit(*this, *_varDecl.value);
-		expectDeposit(numVariables, heightAtStart);
+		expectDeposit(static_cast<int>(numVariables), static_cast<int>(heightAtStart));
 	}
 	else
 	{
 		m_assembly.setSourceLocation(_varDecl.location);
-		int variablesLeft = numVariables;
+		size_t variablesLeft = numVariables;
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
 
 	m_assembly.setSourceLocation(_varDecl.location);
 	bool atTopOfStack = true;
-	for (int varIndex = numVariables - 1; varIndex >= 0; --varIndex)
+	for (size_t varIndex = numVariables; varIndex--; )
 	{
 		YulString varName = _varDecl.variables[varIndex].name;
```

Comment:
Don't you have to replace this by `varIndex - 1` now?

Reply:
I hope this results in a test failure, too...

Reply:
If we replace `vaIndex` anywy, maybe it's better to use a "standard" ascending for loop here and replace it by `numVariables - varIndex - 1`

Reply:
It will not fail, because it is first evaluating i if it is not zero, than decreases, so in loop we will have values that we needed. This is a bit tricky anyway, is better to change if @hrkrshnn is fine with that

Reply:
This was what [SO](https://stackoverflow.com/questions/5458204/unsigned-int-reverse-iteration-with-for-loops) recommended for reversed-unsigned-loop. I'm okay with changing it to range.

Reply:
Maybe something like `boost::irange(numVariables, 0, -1)`?

Reply:
```cpp
for (auto i: boost::irange(size_t(0), numVariables) | boost::adaptors::reversed)`
```

@mijovic this works for me.

Reply:
Solved like that.
Apparently, it wasn't working with
`boost::irange(numVariables, 0, -1)`